storybookjs/storybook
# #6975 lib/ui TS migration - 1

Issue: #6975 (do not close)

## What I did

- Move Sidebar components in the lib/ui into TypeScript

## How to test

- Let CI to run test to see if regressions happened 

related #6975.

---

Along the way to make this PR, I made several `FIXME` comments hope someday someone or myself can pick these up. These changes is laying out of the domain of this package so I did not cleanup these for the moment, otherwise I feel there are endless digging and really overload what I really want to finish in this issue.

My personal experiences/feeling on TypeScript is very positive, but I do feel we need more love on making things really typed, not just looks typed...  (don't want just to abuse `any`...) And I know this can be hard to do, so virtually never going to be the major focus I guess, that is why I need to specifically bring this to the table.

Finally, I found `emotion` brings too many different paradigms on writing CSS-in-JS, so some typing is always not "right" for many reasons. Maybe we could consider to migrate (back) to `styled-components` (and use `stylelint` get css checked)?  This could also make the storybook design system more easier to be integrated in the future.
